sphinx/source/Concepts.rst

```diff
@@ -0,0 +1,167 @@
+
+The identifer contains:
+
+* The parameter group number (PGN)
+
+   * This is what identifies what's in the data paylod, like the subject line of an email
+   * You will need to know some of these (or create some) to communicate with most devices
+   * Some standard ones can be found here: https://www.isobus.net/isobus/pGNAndSPN/?type=PGN
+
+* The source address
+
+   * These describes who is sending the message, like the return address on a peice of mail
+
+* The destination address
+
+   * Some messages allow you to specify a Destination
+   * Some messages are meant to be received by everyone (a broadcast)
+   * The Parameter group number determines if a message is a broadcast or destined for a specific address.
+
+* A priority
+
+   * This determines, when messages collide on the bus, which one "wins" and which transmitter has to re-transmit their message later
+   * This is handled by hardware, not software generally, so we usually only care about this when transmitting messages, not receiveing them
+
+In order to send messages on a J1939 or ISO bus, you need a 1 byte address, which identifes you on the bus.
+Likewise, you need to know the address of who you're sending a mesage to (unless you are sending a broadcast).
+Broadcasts are always sent to the broadcast address, which is 255 (0xFF).
+There's also something called the NULL address (0xFE), which is the address you use when you have no address (or are in the process of getting one, though all this is handled automatically by the stack).
+
+Control Functions
+--------------------
+
+In reality, addresses are useless for most things other than filling in or decoding a CAN frame.
+They tell you nothing about the identity of that device, what it does, who made it, or if it even cares that you are talking to it.
+It's not even reliable for sending messages, as J1939 and ISOBUS devices can change address at any time.
+
+Thus, J1939 and ISO 11783 defined the basic component of a network as a *control function*.
+
+A control function has an address, but is first and foremost identified by its NAME.
+
+A control function's address could change at *any time* via a process called address claiming and arbitration. We'll talk more about this later.
+
+The NAME
+^^^^^^^^^
+
+Conceptually, a NAME is a 64 bit value that uniquely identifies a control function on the network.
+
+A NAME has the following components in it:
+
+* Identity Number
+
+   * This is usually the serial number of the control function.
+   * It's supposed to be unique amongst control functions with identical other NAME values
+
+* Manufacturer Code
+
+   * Identifies who made the control function - based on a list https://www.isobus.net/isobus/manufacturerCode
+
+* ECU Instance
+
+   * Usually increments in NAME order with similar control functions
+   * Basically if you have multiple devices with similar names, this can be used to establish a heierarchy within the ISO NAME amongst them
```

Comment:
```suggestion
   * Basically if you have multiple devices with similar names, this can be used to establish a hierarchy within the ISO NAME amongst them
```